pkg/odo/cli/component/create.go

```diff
@@ -942,109 +936,138 @@ func (co *CreateOptions) downloadStarterProject(projectPassed string, interactiv
+		if err != nil {
+			// we don't need to return (fail) if this happends
+			log.Warning("Unable to delete .git from cloned starter project")
+			downloadSpinner.End(true)
+		}
+		downloadSpinner.End(true)
+
+	} else if starterProject.Zip != nil {
+		url := starterProject.Zip.Location
+		logUrl := starterProject.Zip.Location
+		sparseDir := starterProject.Zip.SparseCheckoutDir
+		downloadSpinner := log.Spinnerf("Downloading starter project %s from %s", starterProject.Name, logUrl)
+		err := checkoutProject(sparseDir, url, path)
+		if err != nil {
+			downloadSpinner.End(false)
 			return err
 		}
-		logUrl = project.Github.Location
-		sparseDir = project.Github.SparseCheckoutDir
-	} else if project.Zip != nil {
-		url = project.Zip.Location
-		logUrl = project.Zip.Location
-		sparseDir = project.Zip.SparseCheckoutDir
+		downloadSpinner.End(true)
 	} else {
 		return errors.Errorf("Project type not supported")
 	}
 
-	log.Info("\nStarter Project")
-	downloadSpinner := log.Spinnerf("Downloading starter project %s from %s", project.Name, logUrl)
-	err = checkoutProject(sparseDir, url, path)
-
-	if err != nil {
-		downloadSpinner.End(false)
-		return err
-	}
-
-	downloadSpinner.End(true)
 	return nil
 }
 
 // Run has the logic to perform the required actions as part of command
 func (co *CreateOptions) Run() (err error) {
 	if experimental.IsExperimentalModeEnabled() {
 		if !co.forceS2i && co.devfileMetadata.devfileSupport {
+			var devfileData []byte
 			// Use existing devfile directly from --devfile flag
 			if co.devfileMetadata.devfilePath.value != "" {
 				if co.devfileMetadata.devfilePath.protocol == "http(s)" {
 					// User specify devfile path is http(s) URL
-					params := util.DownloadParams{
-						Request: util.HTTPRequestParams{
-							URL:   co.devfileMetadata.devfilePath.value,
-							Token: co.devfileMetadata.token,
-						},
-						Filepath: DevfilePath,
+					params := util.HTTPRequestParams{
+						URL:   co.devfileMetadata.devfilePath.value,
+						Token: co.devfileMetadata.token,
 					}
-					err = util.DownloadFile(params)
+					devfileData, err = util.DownloadFileInMemory(params)
 					if err != nil {
 						return errors.Wrapf(err, "failed to download devfile for devfile component from %s", co.devfileMetadata.devfilePath.value)
 					}
 				} else if co.devfileMetadata.devfilePath.protocol == "file" {
-					// User specify devfile path is file system link
-					info, err := os.Stat(co.devfileMetadata.devfilePath.value)
+					devfileData, err = ioutil.ReadFile(co.devfileMetadata.devfilePath.value)
 					if err != nil {
-						return err
+						return errors.Wrapf(err, "failed to read devfile from %s", co.devfileMetadata.devfilePath)
 					}
-					err = util.CopyFile(co.devfileMetadata.devfilePath.value, DevfilePath, info)
+				}
+			} else {
+				if util.CheckPathExists(DevfilePath) {
+					// if local devfile already exists read that
+					// odo create command was exected in a directory already containing devfile
```

Comment:
expected

Reply:
this was supposed to be executed ;-)
fixed